### ISSUE TITLE
Memory leak with gauge field caching for MILC

### ISSUE THREAD

**Issue by mathiaswagner** [lattice/quda#479]
Citing Carletons mail:

> We immediately ran into a problem with one of the job steps.  The job step calculated two clover propagators, then one staggered propagator, and then ran out of memory on 32 GPUs while calculating a second staggered propagator.  I understand that with quda-0.8.0 there is a mechanism for keeping the gauge links in GPU memory between calls.

From a run on 64 GPUs.

```
Device memory used = 2916.7 MB
Page-locked host memory used = 848.1 MB
Total host memory used >= 996.6 MB

WARNING: The following internal memory allocations were not freed.

Type    Pointer          Size             Location
----------------------------------------------------------
Device      0xb054e0000        264241152  cudaGaugeField(), /N/u/sg/BigRed2/quda-0.8.0/quda-0.8.0/lib/cuda_gauge_field.cu:42
Device      0xb42da0000         37748736  create(), /N/u/sg/BigRed2/quda-0.8.0/quda-0.8.0/lib/cuda_color_spinor_field.cu:173
Host         0x3a1d0520             2064  operator new(), /N/u/sg/BigRed2/quda-0.8.0/quda-0.8.0/include/object.h:23
Host         0x42bec900              896  operator new(), /N/u/sg/BigRed2/quda-0.8.0/quda-0.8.0/include/object.h:23
```

**Comment by maddyscientist** [comment]
This bug should now be fixed with #303f60e67340184647fcdb6a0bed6539d94bce57, though it needs full verification.

**Comment by detar** [comment]
Thank you!  Do you need anything more from us?

Carleton
